File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution
{
public:
    int evalRPN(vector<string> &tokens)
    {
        stack<long> st;

        for (int i = 0; i < tokens.size(); i++)
        {
            if (tokens[i].size() == 1 && ispunct(tokens[i][0]))
            {
                long x = st.top();
                st.pop();
                long y = st.top();
                st.pop();
                if (tokens[i] == "+")
                    st.push(x + y);
                else if (tokens[i] == "-")
                    st.push(y - x);
                else if (tokens[i] == "*")
                    st.push(x * y);
                else if (tokens[i] == "/")
                    st.push(y / x);
            }
            else
            {
                long num = stoi(tokens[i]);
                st.push(num);
            }
        }

        return st.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    String operators = "+-/*";
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        for (String value : tokens) {

            if (isOperator(value)){
                calcOperation(stack,value);
            }
            else
            {
                stack.push(Integer.parseInt(value));
            }

        }

        return stack.pop();


    }

    private void calcOperation(Stack<Integer> stack, String value) {
        int number1 = stack.pop();
        int number2 = stack.pop();
        if (value.equals("+")) {
            stack.add(number1 + number2);
        } else if (value.equals("-")) {
            stack.add(number2 - number1);
        } else if (value.equals("*")) {
            stack.add(number1 * number2);
        } else  {

            stack.add(number2 / number1);
        }
    }

    private boolean isOperator(String value) {
        return operators.contains(value);
    }
}